pkg/controller/controller_utils.go

```diff
@@ -1045,3 +1051,48 @@ func ComputeHash(template *v1.PodTemplateSpec, collisionCount *int32) string {
 
 	return rand.SafeEncodeString(fmt.Sprint(podTemplateSpecHasher.Sum32()))
 }
+
+func AddOrUpdateLabelsOnNode(kubeClient clientset.Interface, nodeName string, labelsToUpdate map[string]string) error {
```

Comment:
Is it possible to combine "the update call to API server" with `TaintByCondition` ? If we want to it later because of 1.14 schedule, please help to add a TODO or open an issue for that :)

Reply:
The `TaintByCondition` mechanism is more complex (i.e., swapping taints) and is still gated. I think merging them may simply cause unnecessary bugs. Label reconciling happens rarely (e.g., node upgrade/downgrade with mismatching OS/arch) , and having a separate apiserver update seems okay to me.

Reply:
I'm ok to keep it as is.

we can have seperate PR for that; as prefer to have such logic : get node info, reconcile it (e,g. labels, conditions), update it back to api server. If we want to add other reconcile logic, we only need to enhance the seconds phase.

Reply:
Sounds good.